src/viam/app/app_client.py

```diff
@@ -400,12 +491,50 @@ async def get_organization_namespace_availability(self, public_namespace: str) -
         return response.available
 
     async def update_organization(
-        self, name: Optional[str] = None, public_namespace: Optional[str] = None, region: Optional[str] = None
+        self,
+        name: Optional[str] = None,
+        public_namespace: Optional[str] = None,
+        region: Optional[str] = None,
+        cid: Optional[str] = None,
     ) -> Organization:
-        raise NotImplementedError()
+        """Updates organization details.
 
-    async def delete_organization(self, organization_id: str) -> None:
-        raise NotImplementedError()
+        Args:
+            name (Optional[str]): If provided, updates the org's name.
+            public_namespace (Optional[str]): If provided, sets the org's namespace if it hasn't already been set.
+            region (Optional[str]): If provided, updates the org's region.
+            cid (Optional[str]): If provided, update's the org's CRM ID.
+
+        Raises:
+            GRPCError: If the org's namespace has already been set, or if the provided namespace is already taken.
+
+        Returns:
+            viam.proto.app.Organization: The updated organization.
+        """
+        organization_id = await self._get_organization_id()
+        request = UpdateOrganizationRequest(
+            organization_id=organization_id,
+            public_namespace=public_namespace,
+            region=region,
+            cid=cid,
+            name=name,
+        )
+        response: UpdateOrganizationResponse = await self._app_client.UpdateOrganization(request, metadata=self._metadata)
+        return response.organization
+
+    async def delete_organization(self, org_id: Optional[str] = None) -> None:
+        """Deletes an organization.
+
+        Args:
+            org_id (Optional[str]): ID of the organization to delete. If None, defaults to the
+            currently-authed org.
+
+        Raises:
+            GRPCError: if the organization is not authed to, or has locations in it.
+        """
+        organization_id = org_id if org_id else await self._get_organization_id()
```

Comment:
same-ish Q as noted above

```suggestion
        organization_id = org_id if org_id is not None else await self._get_organization_id()
```